File: serve4j/src/main/java/billow/fun/serve4j/model/Post.java
package billow.fun.serve4j.model;

import billow.fun.serve4j.BaseEntity;

import javax.persistence.Column;
import javax.persistence.Entity;

@Entity
public class Post extends BaseEntity {
    private String title;
    @Column(columnDefinition = "LONGTEXT")
    private String content;
    private String category;
    private String[] fileList;

    public Post() {
    }

    public Post(String title, String content, String category, String[] fileList) {
        this.title = title;
        this.content = content;
        this.category = category;
        this.fileList = fileList;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String[] getFileList() {
        return fileList;
    }

    public void setFileList(String[] fileList) {
        this.fileList = fileList;
    }
}

File: server/src/main/kotlin/billow/fun/cms/server/model/Post.kt
package billow.`fun`.cms.server.model

import billow.`fun`.cms.server.BaseEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.repository.query.Param
import javax.persistence.Column
import javax.persistence.Entity

@Entity
data class Post(
        val title: String,
        @Column(columnDefinition = "LONGTEXT")
        val content: String,
        val category: String,
        val fileList: Array<String>
) : BaseEntity() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Post

        if (title != other.title) return false
        if (content != other.content) return false

        return true
    }

    override fun hashCode(): Int {
        var result = title.hashCode()
        result = 31 * result + content.hashCode()
        return result
    }
}


interface PostRepo : JpaRepository<Post, Long> {
    fun findByCategory(@Param("category") category: String): Collection<Post>
}
